lib/src/solver/report.dart

```diff
@@ -130,6 +130,39 @@ class SolveReport {
     }
   }
 
+  /// Displays a two-line message, number of outdated packages and an
+  /// instruction to run `pub outdated` if outdated packages are detected.
+  void reportOutdated() {
+    var outdatedPackagesCount = 0;
+
+    final names = _result.packages.map((id) => id.name);
+    for (var name in names) {
+      var newId = _dependencies[name];
+
+      // See if there are any newer versions of the package that we were
+      // unable to upgrade to.
+      if (newId != null) {
```

Comment:
When and why can this null?

Reply:
I thought the may not exist on the map, but I think I am wrong. I was previously following this logic:
https://github.com/dart-lang/pub/blob/572a23f815a5d632a8c2b259b691792674cd8b4d/lib/src/solver/report.dart#L205-L221